service/storage_service.cc

```diff
@@ -3347,8 +3352,8 @@ storage_service::get_splits(const sstring& ks_name, const sstring& cf_name, rang
 };
 
 dht::token_range_vector
-storage_service::get_ranges_for_endpoint(const sstring& name, const gms::inet_address& ep) const {
-    return _db.local().find_keyspace(name).get_effective_replication_map()->get_ranges(ep);
+storage_service::get_ranges_for_endpoint(const locator::effective_replication_map_ptr& erm, const gms::inet_address& ep) const {
+    return erm->get_ranges(ep);
 }
```

Comment:
Really the function can just go away, no?

Or you could make it static to emphasize it's pure and doesn't depend on any state outside the input.

Reply:
I'd like to reduce the churn, so making it static sounds good.